src/main/java/org/jenkinsci/plugins/docker/workflow/WithContainerStep.java

```diff
@@ -124,7 +124,7 @@ private static void destroy(String container, Launcher launcher, Node node, EnvV
 
         @Override public boolean start() throws Exception {
             EnvVars envReduced = new EnvVars(env);
-            EnvVars envHost = computer.getEnvironment();
+            EnvVars envHost = computer.buildEnvironment(TaskListener.NULL);
```

Comment:
I don't get it. `buildEnvironment` do return environment to run commands on the node, doesn't it ? And docker-pipeline do run `docker` CLI on the node. So what ?
`envReduced` is about running things inside a container on the node, which is yet another story